components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/service/CertificateValidationManagementService.java

```diff
@@ -0,0 +1,124 @@
+     */
+    List<Validator> getValidators(int tenantId) throws X509ConfigurationException;
+
+    /**
+     * Get the validator by name.
+     *
+     * @param name     Name of the validator.
+     * @param tenantId Tenant Id.
+     * @return Validator.
+     * @throws X509ConfigurationException If an error occurs while getting the validator.
+     */
+    Validator getValidator(String name, int tenantId) throws X509ConfigurationException;
+
+    /**
+     * Update the validator.
+     *
+     * @param validator Validator.
+     * @param tenantId  Tenant Id.
+     * @return Updated validator.
+     * @throws X509ConfigurationException If an error occurs while updating the validator.
+     */
+    Validator updateValidator(Validator validator, int tenantId) throws X509ConfigurationException;
+
+    /**
+     * Get CA Certificates.
+     *
+     * @param tenantId Tenant Id.
+     * @return List of CA certificates.
+     * @throws X509ConfigurationException If an error occurs while getting the CA certificates.
+     */
+    List<CACertificateInfo> getCACertificates(int tenantId) throws X509ConfigurationException;
+
+    /**
+     * Add CA Certificate.
+     *
+     * @param caCertificate CA Certificate.
+     * @param tenantId      Tenant Id.
+     * @return Added CA Certificate Info.
+     * @throws X509ConfigurationException If an error occurs while adding the CA certificate.
+     */
+    CACertificate addCACertificate(X509Certificate caCertificate, int tenantId) throws X509ConfigurationException;
+
+    /**
+     * Get CA Certificate.
+     *
+     * @param certificateId Certificate Id.
+     * @param tenantId      Tenant Id.
+     * @return CA Certificate Info.
+     * @throws X509ConfigurationException If an error occurs while getting the CA certificate.
+     */
+    CACertificateInfo getCaCertificate(String certificateId, int tenantId) throws X509ConfigurationException;
+
+    /**
+     * Update CA Certificate.
+     *
+     * @param certificateId Certificate Id.
+     * @param certificate   CA Certificate.
+     * @param tenantId      Tenant Id.
+     * @return Updated CA Certificate Info.
+     * @throws X509ConfigurationException If an error occurs while updating the CA certificate.
+     */
+    CACertificateInfo updateCACertificate(String certificateId, X509Certificate certificate, int tenantId)
+            throws X509ConfigurationException;
+
+    /**
+     * Delete CA Certificate.
+     *
+     * @param certificateId Certificate Id.
+     * @param tenantId      Tenant Id.
+     * @throws X509ConfigurationException If an error occurs while deleting the CA certificate.
+     */
+    void deleteCACertificate(String certificateId, int tenantId) throws X509ConfigurationException;
+
+    /**
+     * Add default validator configurations to the storage.
+     *
+     * @param tenantDomain Tenant Domain.
+     */
+    void addDefaultValidationConfigInRegistry(String tenantDomain);
```

Comment:
`addDefaultValidationConfigInRegistry` -> `addDefaultValidationConfigs`
Lets avoid having `InRegistry` part in the service level.

Reply:
ack

Reply:
fixed with https://github.com/wso2-extensions/identity-x509-commons/pull/57/commits/52f92488d265780c25c0d23c5ec638ca00bd3f59